src/java.base/share/classes/java/util/zip/InflaterInputStream.java

```diff
@@ -124,13 +124,21 @@ public int read() throws IOException {
     }
 
     /**
-     * Reads uncompressed data into an array of bytes. If {@code len} is not
-     * zero, the method will block until some input can be decompressed; otherwise,
-     * no bytes are read and {@code 0} is returned.
+     * Reads uncompressed data into an array of bytes, returning the number of inflated
+     * bytes. If {@code len} is not zero, the method will block until some input can be
+     * decompressed; otherwise, no bytes are read and {@code 0} is returned.
+     * <p>
+     * If <i>n</i> denotes a nonzero number of inflated bytes to return, then {@code b[off]}
+     * through {@code b[off+}<i>n</i>{@code -1]} will contain the uncompressed data. The
+     * elements {@code b[off+}<i>n</i>{@code ]} through {@code b[off+}<i>len</i>{@code -1]}
+     * are undefined (an implementation is free to change them during the inflate
+     * operation). If the return value is -1 or an exception is thrown, then the content of
+     * {@code b[off]} to {@code b[off+}<i>len</i>{@code -1]} is undefined.
```

Comment:
Consider this more precise wording, both here and in `GZIPInputStream`, `ZipInputStream`, and `JarInputStream`:
```
     * If this method returns a nonzero integer <i>n</i> then {@code buf[off]}
     * through {@code buf[off+}<i>n</i>{@code -1]} contain the uncompressed
     * data.  The content of elements {@code buf[off+}<i>n</i>{@code ]} through
     * {@code buf[off+}<i>len</i>{@code -1]} is undefined, contrary to the
     * specification of the {@link java.io.InputStream InputStream} superclass,
     * so an implementation is free to modify these elements during the inflate
     * operation. If this method returns {@code -1} or throws an exception then
     * the content of {@code buf[off]} through {@code buf[off+}<i>len</i>{@code
     * -1]} is undefined.
```